### PR TITLE
Anastasia DL Ch1,2,3,4 finished

### PR DESCRIPTION
**Pull request by AnastasiaProkaieva**

### CODE REVIEW

File: Chapter_2_Designing_Databricks_Day_One/Project: MultiClass classification for Computer Vision /CH2-01-Downloading_Images.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# MAGIC %run ../../global-setup $project_name=cv_clf
+
+# COMMAND ----------
+
+# DBTITLE 1,Add your credentials to connect to the Kaggle Account 
+import os
+
+os.environ[
+    "kaggle_username"
+] = dbutils.secrets.get("mlaction", "kaggle_name")  # replace with your own credential here temporarily or set up a secret scope with your credential
```
Comment (stephanieamrivera) [inline Chapter_2_Designing_Databricks_Day_One/Project: MultiClass classification for Computer Vision /CH2-01-Downloading_Images.py:27]: We should unify this with the kaggle download for favorita

File: Chapter_3_Building_Out_Our_Bronze_Layer/Project: MultiClass classification for Computer Vision /Ch3-01-Loading_Images_2_DeltaTables.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            df.write.format("delta").mode("append").save("{path2write}{name2write}")
+        if returnDF:
+            return df
+
+# COMMAND ----------
+
+# MAGIC %md 
+# MAGIC #### Note 
+# MAGIC
+# MAGIC  As of now DeltaTorch Loader requires you to have tables in delta format under Volumes when UC is used in the future the functionality will involve. It's simply due to some security permissions on the read/write that UC has. <br>
```
Comment (stephanieamrivera) [inline Chapter_3_Building_Out_Our_Bronze_Layer/Project: MultiClass classification for Computer Vision /Ch3-01-Loading_Images_2_DeltaTables.py:70]: Great note. Lets add it to the text of the book rather than the code :)

File: Chapter_2_Designing_Databricks_Day_One/Project: MultiClass classification for Computer Vision /CH2-01-Downloading_Images.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+# COMMAND ----------
+
+# DBTITLE 1,Unzip your data under Volumes 
+!mkdir /Volumes/{catalog}/{database_name}/files/intel_image_clf/
+!mkdir /Volumes/{catalog}/{database_name}/files/intel_image_clf/raw_images
+
+# this can take up to few hours  
+# or load a few examples to the UC on your own if the time is a constraint 
+!unzip -n /local_disk0/intel-image-classification.zip -d /Volumes/{catalog}/{database_name}/files/intel_image_clf/raw_images
```
Comment (stephanieamrivera) [inline Chapter_2_Designing_Databricks_Day_One/Project: MultiClass classification for Computer Vision /CH2-01-Downloading_Images.py:49]: the volumes file path should be available via the global script

File: Chapter_3_Building_Out_Our_Bronze_Layer/Project: MultiClass classification for Computer Vision /Ch3-01-Loading_Images_2_DeltaTables.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+# COMMAND ----------
+
+# MAGIC %sql 
+# MAGIC OPTIMIZE delta.`/Volumes/$catalog/$database_name/intel_image_clf/train_imgs_main.delta`
+
+# COMMAND ----------
+
+# MAGIC %sql 
+# MAGIC ALTER TABLE delta.`/Volumes/$catalog/$database_name/intel_image_clf/train_imgs_main.delta` SET TBLPROPERTIES ('delta.enableDeletionVectors' = false);
```
Comment (stephanieamrivera) [inline Chapter_3_Building_Out_Our_Bronze_Layer/Project: MultiClass classification for Computer Vision /Ch3-01-Loading_Images_2_DeltaTables.py]: Please explain in the chapter text why we want to disable deletion vectors

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py [hunk 5]
```diff
@@ -0,0 +1,379 @@
+# Databricks notebook source
+!pip install pytorch-lightning==2.1.2 deltalake==0.14.0 deltatorch==0.0.3 evalidate==2.0.2 pillow==10.1.0
```
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py:2]: should we make requirements.txt for these? I haven't done it yet for mine either

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+    def val_dataloader(self):
+        return self.dataloader(val_delta_path, batch_size=100)
+
+    def test_dataloader(self):
+        return self.dataloader(val_delta_path, batch_size=50)
+
+# COMMAND ----------
+
+# MAGIC %md 
+# MAGIC ### Native Torch Loader
```
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py:194]: maybe separate this to another notebook and link it?

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+# MAGIC         return DataLoader(self.train_dir, batch_size=50, sampler=self.valid_sampler, num_workers=2)
+# MAGIC
+# MAGIC     def test_dataloader(self):
+# MAGIC         return DataLoader(self.test_data, batch_size=32, sampler=None, num_workers=2)
+# MAGIC
+# MAGIC ```
+
+# COMMAND ----------
+
+#Check GPU availability
```
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py:259]: nice!

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+                    pip_requirements=reqs,
+                )
+                mlflow.set_tag("ml2action", "cv_uc")
+
+
+# COMMAND ----------
+
+train_distributed(MAX_EPOCH_COUNT, STRATEGY)
+
+# COMMAND ----------
```
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-01-Training_BaseModel.py:365]: another notebook?

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-02-Scoring_BestModel.py
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-02-Scoring_BestModel.py:1]: I think this should be ch 7

File: Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-02-Scoring_BestModel.py [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+    index = pred_class_func(img)
+    plt.subplot(5,5,i+1)
+    plt.title(classes[index])
+    plt.axis('off')
+    plt.imshow(img)
+
+# COMMAND ----------
+
+# MAGIC %md 
+# MAGIC
```
Comment (stephanieamrivera) [inline Chapter_6_Searching_For_Signal/Project: MultiClass classification for Computer Vision /Ch6-02-Scoring_BestModel.py:109]: another notebook

**Review by TheDataCurator — APPROVED** [review]

**Review by TheDataCurator — APPROVED** [review]

File: global-setup.py [hunk 10]
```diff
@@ -17,19 +17,24 @@
...
 assert project_name in possible_projects, "project_name unknown, did you type correctly? You can add new projects to the list in the setup file."
 
+
 # VERIFY DATABRICKS VERSION COMPATIBILITY ----------
 try:
   min_required_version = dbutils.widgets.get("min_dbr_version")
 except:
   min_required_version = "13.0"
 
+if project_name in ["rag_chatbot", "cv_clf"]:
```
Comment (stephanieamrivera) [inline global-setup.py:35]: we can require 14 across the board

File: global-setup.py [hunk 11]
```diff
@@ -113,8 +118,52 @@ def use_and_create_db(catalog, database_name):
...
-# import os
 
-# os.environ['kaggle_username'] = dbutils.secrets.get("lakehouse-in-action", "kaggle_username")
 
-# os.environ['kaggle_key'] = dbutils.secrets.get("lakehouse-in-action", "kaggle_key")
+# COMMAND ----------
+
+if project_name == "cv_clf":
+  try:
+    print("You are required to set your token and host if you want to use MLFlow tracking while using DDP \n")
```
Comment (stephanieamrivera) [inline global-setup.py:127]: MLflow rather than MLFlow

File: global-setup.py [hunk 12]
```diff
@@ -113,8 +118,52 @@ def use_and_create_db(catalog, database_name):
...
+        .notebook()
+        .getContext()
+        .extraContext()
+        .apply("api_url")
+    )
+    db_token = (
+        dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiToken().get()
+    )
+  except:
+    print("Your MLFLow logging may not function correctly due to the missing db_host and db_token variables \n")
```
Comment (stephanieamrivera) [inline global-setup.py:141]: MLflow

File: global-setup.py [hunk 13]
```diff
@@ -113,8 +118,52 @@ def use_and_create_db(catalog, database_name):
...
+
+
+if project_name == "cv_clf": 
+
+  MAIN_DIR_UC = f"/Volumes/{catalog}/{database_name}/files/intel_image_clf/raw_images"
```
Comment (stephanieamrivera) [inline global-setup.py:146]: lets use the volume_file_path

**Review by stephanieamrivera — APPROVED** [review]
I left comments. I will still approve and merge. feel free to poke me about my comments
